packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1234,10 +1234,13 @@ abstract class FlutterCommand extends Command<void> {
       }
     }
 
+    final String? flavor = argParser.options.containsKey('flavor') ? stringArg('flavor') : null;
+    if (flavor != null) {
+      dartDefines.add('FLUTTER_APP_FLAVOR=$flavor');
```

Comment:
have we already parsed user-provided dart defines at this point, and then can we check for it?